lib/vscode/src/vs/workbench/api/common/shared/webview.ts

```diff
@@ -55,9 +55,12 @@ export function asWebviewUri(
 		});
 	}
 
+	// NOTE@coder: Add the port separately because if the port is in the domain the
+	// URL will be invalid and the browser will not request it.
+	const authorityUrl = new URL(`http://${resource.authority}`);
 	return URI.from({
 		scheme: Schemas.https,
-		authority: `${resource.scheme}+${resource.authority}.${webviewRootResourceAuthority}`,
+		authority: `${resource.scheme}+${authorityUrl.hostname}.${webviewRootResourceAuthority}${authorityUrl.port ? (':' + authorityUrl.port) : ''}`,
```

Comment:
@jsjoeio IMO the bug stems from modifying URLs as strings, rather than leaning on the browser’s innate URL constructor Would something like this help with URL?
```ts
// e.g. new URL(‘abc://localhost:3000/foo/bar?baz=qux')
// The scheme isn’t too big of a deal here but it helps to lean on what’s known from above.
const authorityUrl = new URL(`${resource.scheme}://${resource.authority}`);

authorityUrl.hostname = `${authorityUrl.hostname}+${resource.authority}.${webviewRootResourceAuthority}`;
// Remove the protocol
const authority = authorityUrl.toString().slice(`${authorityUrl.protocol}//`.length);

return URI.from({
  scheme: Schemas.https,
  authority,
  path: resource.path,
  fragment: resource.fragment,
  query: resource.query,
});
```

Reply:
Converting to draft stage and then i'll re-request a review

Reply:
Should we `try/catch` the URL construction?

Reply:
Sure! Good call!

Reply:
Okay...looking at your solution, I don't think I fully understand how it helps. I tried it, but I think either the hostname or the port gets mangled (though I might be doing something else wrong). I'm going to leave as is for now. And if you want to follow-up with more details or have me submit another PR to improve this, happy to do so!